dockstore-webservice/src/main/java/io/dockstore/webservice/resources/SourceFileResource.java

```diff
@@ -0,0 +1,96 @@
+package io.dockstore.webservice.resources;
+
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+
+import com.codahale.metrics.annotation.Timed;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.core.SourceFile;
+import io.dockstore.webservice.core.Tag;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.WorkflowVersion;
+import io.dockstore.webservice.jdbi.TagDAO;
+import io.dockstore.webservice.jdbi.WorkflowVersionDAO;
+import io.dropwizard.auth.Auth;
+import io.dropwizard.hibernate.UnitOfWork;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.Authorization;
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
+import io.swagger.v3.oas.annotations.security.SecurityScheme;
+import io.swagger.v3.oas.annotations.security.SecuritySchemes;
+import org.apache.http.HttpStatus;
+import org.hibernate.SessionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.dockstore.webservice.Constants.JWT_SECURITY_DEFINITION_NAME;
+
+@Path("/versions")
+@Api("versions")
+@Produces(MediaType.APPLICATION_JSON)
+@SecuritySchemes({ @SecurityScheme (type = SecuritySchemeType.HTTP, name = "bearer", scheme = "bearer")})
+@io.swagger.v3.oas.annotations.tags.Tag(name = "sourceFile", description = ResourceConstants.SOURCEFILE)
+public class SourceFileResource {
+    private static final Logger LOG = LoggerFactory.getLogger(SourceFileResource.class);
+
+    private final SessionFactory sessionFactory;
+    private final TagDAO tagDAO;
+    private final WorkflowVersionDAO workflowVersionDAO;
+
+
+    public SourceFileResource(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+        this.tagDAO = new TagDAO(sessionFactory);
+        this.workflowVersionDAO = new WorkflowVersionDAO(sessionFactory);
+
+    }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("{versionId}/sourcefiles")
+    @ApiOperation(value = "Retrieve sourcefiles for an entry version", authorizations = {
```

Comment:
This should use OpenAPI annotations instead.  I believe there's testing utilities for openapi client so you don't have to worry about that part.